src/auth-strategies/azure_ad_strategy.ts

```diff
@@ -59,12 +59,19 @@ export const setupBearerStrategy = (
       (
         _: express.Request,
         profile: AdUser,
-        done: (err: Error | undefined, profile?: AdUser) => void
+        done: (
+          err: Error | undefined,
+          user?: AdUser & { readonly kind: "azure-ad" }
+        ) => void
       ) => {
-        return cb(profile.oid, profile)
+        const user = {
+          ...profile,
+          kind: "azure-ad" as const /* to define which strategy the user is extracted by */
```

Comment:
If you have an Enum for Available Strategy:
```suggestion
        const user = {
          ...profile,
          kind: AvailableStrategyEnum.AZURE /* to define which strategy the user is extracted by */
        };
```